config/mtls.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+
+	"github.com/pomerium/pomerium/pkg/cryptutil"
+	"github.com/pomerium/pomerium/pkg/grpc/config"
+)
+
+// DownstreamMTLSSettings specify the downstream client certificate requirements.
+type DownstreamMTLSSettings struct {
+	// CA is the base64-encoded certificate authority (or bundle of certificate
+	// authorities) that should serve as the trust root(s). These will be
+	// advertised in the initial TLS handshake.
+	CA string `mapstructure:"ca" yaml:"ca"`
+
+	// CAFile is the path to a file containing the certificate authority (or
+	// bundle of certificate authorities) that should serve as the trust
+	// root(s). These will be advertised in the initial TLS handshake.
+	CAFile string `mapstructure:"ca_file" yaml:"ca_file"`
+
+	// CRL is the base64-encoded certificate revocation list (or bundle of
+	// CRLs) to use when validating client certificates.
+	CRL string `mapstructure:"crl" yaml:"crl,omitempty"`
+
+	// CRLFile is the path to a file containing the certificate revocation
+	// list (or bundle of CRLs) to use when validating client certificates.
+	CRLFile string `mapstructure:"crl_file" yaml:"crl_file,omitempty"`
+}
+
+// GetCA returns the certificate authority (or nil if unset).
+func (s *DownstreamMTLSSettings) GetCA() ([]byte, error) {
+	if s.CA != "" {
+		return base64.StdEncoding.DecodeString(s.CA)
+	}
+	if s.CAFile != "" {
+		return os.ReadFile(s.CAFile)
```

Comment:
I don't remember whether it was during audit or not, but mostly we wrap reading from sources into LimitReader just in case we're accidentally pointed to some large / infinite file. i.e. https://github.com/pomerium/pomerium/blob/bb2fb9730ad4f714d025578305236081fba32db9/pkg/cryptutil/certificates.go#L235